OpenRA.Game/Effects/IEffect.cs

```diff
@@ -18,5 +19,6 @@ public interface IEffect
 	{
 		void Tick(World world);
 		IEnumerable<IRenderable> Render(WorldRenderer r);
+		Rectangle VisualBounds(WorldRenderer wr);
```

Comment:
If you pass the bounds into the ScreenMap.Add call then you won't need this on the interface, and can avoid having the dummy implementations on the effects that don't interact with the ScreenMap.